201-204 project management and software development

edvinskis opened the pull request:
Addresses:
- #168 
- #169
- #201 
- #202 
- #203 
- #204

mwakok commented:
Hi Elviss, thanks for all the work. I have some suggestions for further improves:

- The project management sections now introduces topics before they are explained, for example we now have a workflow for using issues, before the explanation of issues.
- Consider splitting the `project_organisation` section into `project_structure` and `reusing_projects` (sections of templates and reusing). I feel there's otherwise too much information in a single document.
- We should think about the use (or formatting) of admonitions, the `project_structure` has become somewhat too colorful in my opinion, which detracts for the readability and clarity.
- For branch management, is it possible to use the same color for the branch type across all examples?

edvinskis commented:
Thank you @mwakok for the feedback!

1. Good point! I rephrased a bit and rearranged the order.
2. I've split the project organisation subsection as you suggested. By doing this I feel like this also addresses point number three. However, I agree that we should revisit the formatting of admonitions at some point, perhaps once the research software section content is finalized.
4. Yes! I updated the mermaid diagrams.

mwakok reviewed (approved):
Looks really nice!

aysunrhn commented on docs/software/development_workflow/project_structure.md:
```diff
@@ -0,0 +1,177 @@
+│
+├── .gitignore                # Untracked files 
+├── requirements.txt          # Software dependencies (environment.yml if using Conda)
+│							  #	↑ Even better to use a build system config (pyproject.toml)
+│							  #	↑ which is becoming the new standard
```
It is worth adding a note about using `pyproject.toml` especially when this directory structure is adopted, it is useful to specify the locations of scripts to be tested. We found this to be a more efficient way to define it in a project. 

https://docs.pytest.org/en/stable/explanation/goodpractices.html#choosing-a-test-layout

I could add a small note about this along with this link to the docs.

aysunrhn replied:
Got it, thank you. I will assign myself to the issue, I can contribute to that section.
